packages/api/src/codegen/languages/typescript/index.ts

```diff
@@ -458,25 +397,115 @@ sdk.server('https://eu.api.example.com/v14');`),
-      // `export =` -- only TS types will have this export style.
-      isExportEquals: this.compilerTarget === 'cjs' && this.outputJS,
+      // Because we're exporting `createSDK` as an IIFE constant we need to have it exported as
+      // `export default createSDK`. `addExportAssignment` by default wants it exported as
+      // `export = createSDK`, which will throw TS errors because we may also be exporting types in
+      // the `./types.ts` file.
+      isExportEquals: false,
       expression: 'createSDK',
     });
 
     return sourceFile;
   }
 
+  /**
+   * Create the `tsconfig.json` file that will allow this SDK to be compiled for use.
+   *
+   */
+  createTSConfig() {
+    const sourceFile = this.project.createSourceFile('tsconfig.json', '');
+
+    const config: TsConfigJson = {
+      compilerOptions: {
+        checkJs: true,
+        esModuleInterop: true,
+        module: 'NodeNext',
+        resolveJsonModule: true,
+      },
+      include: ['./src/**/*'],
+      exclude: ['dist'],
+    };
+
+    sourceFile.addStatements(JSON.stringify(config, null, 2));
+
+    return sourceFile;
+  }
+
+  /**
+   * Create the `package.json` file that will ultimately make this SDK available to use.
+   *
+   */
+  createPackageJSON() {
+    const sourceFile = this.project.createSourceFile('package.json', '');
+
+    const hasTypes = !!Object.keys(this.schemas).length;
+
+    const info = this.spec.getDefinition().info;
+    let pkgVersion = semver.coerce(info.version);
+    if (!pkgVersion) {
+      // If the version that's in `info.version` isn't compatible with semver NPM won't be able to
+      // handle it properly so we need to fallback to something it can.
+      pkgVersion = semver.coerce('0.0.0') as SemVer;
+    }
+
+    const tsupOptions: Options = {
+      cjsInterop: true,
+      clean: true,
+      dts: true,
+      entry: [
+        './src/index.ts',
+        // If this SDK has schemas and generated types then we should also export those too so
+        // they're available to use.
+        hasTypes ? './src/types.ts' : '',
+      ].filter(Boolean),
+      format: ['esm', 'cjs'],
+      minify: false,
+      shims: true,
+      sourcemap: true,
+      splitting: true,
+    };
+
+    const dependencies = Object.entries(this.requiredPackages)
+      .map(([dep, { version }]) => ({ [dep]: version }))
+      .reduce((prev, next) => Object.assign(prev, next));
+
+    const pkg: PackageJson = {
+      name: `@api/${this.identifier}`,
+      version: pkgVersion.version,
+      main: './dist/index.js',
+      types: './dist/index.d.ts',
+      module: './dist/index.mts',
+      exports: {
+        '.': {
+          import: './dist/index.mjs',
+          require: './dist/index.js',
+        },
+        ...(hasTypes
+          ? {
+              './types': {
+                import: './dist/types.d.mts',
+                require: './dist/types.d.ts',
+              },
+            }
+          : {}),
+      },
+      dependencies,
```

Comment:
per this comment, might be good to add the dev-deps here so we're not running `npm i` several times